includes/modules/create_account.php

```diff
@@ -110,6 +110,9 @@
 
   if (ACCOUNT_DOB == 'true') {
     if (ENTRY_DOB_MIN_LENGTH > 0 or !empty($_POST['dob'])) {
+      if (preg_match('/^([0-9]{4})(|-|\/)([0-9]{2})\2([0-9]{2})$/', $dob)) {

```

Comment:
The logic here as I'm reading it is "4 digits, plus a delimiter of either `-` or `/`, plus 2 digits, and same delimiter, and 2 digits", right?

Looking at the regex closer, is the first `|` necessary here?
`(|-|\/)`

Reply:
Somewhat correct, the first emptiness before the first pipe (`|`) indicates no divider, such that would be 4 digits, no delimiter, a `-`, or a `/`, 2 digits, and same excluded or included delimiter, and then 2 digits.

Necessary may be a strong statement; however, it allows supporting not having a divider between the 4-2-2 pairing as identified in the php manual ISO8601 section for 3 of the 4 presented methods: http://php.net/manual/en/datetime.formats.date.php

The first pipe (`|`) inspects the entire content as an eight digit number by having no "filter" on it.  Without it, some testing/testers failed the data as invalid.

I thought that trying to incorporate the 4th method would cause an issue as it presents YY-MM-DD which when speaking of historical dates, if any, would be problematic with all date types if say referring to 11-07-10 which could be interpreted in any way because all numbers are less than the maximum number of 12 for month. :/ THAT will be an issue that may have to be addressed later in some other way...

Reply:
Usually to make an item optional, a `?` is used (`?` means "0 or more times").

Would this give the same results?

```diff
-'/^([0-9]{4})(|-|\/)([0-9]{2})\2([0-9]{2})$/'
+'/^([0-9]{4})(-|\/)?([0-9]{2})\2?([0-9]{2})$/'
```

Reply:
Not exactly: the `\2?` makes the 2nd "entry" optional such that the following tests all appear as acceptable (random date chosen but consistently):
2017-04-09 (OK)
20170409 (OK)
2017/04/09 (OK)
2017-0409 (not)
2017/0409 (not)

Reply:
Ya, I wondered about that after posting. Thanks for testing that possibility.

Reply:
No problem, appreciate the review especially when it comes to the wild animal of regex.